### PR TITLE
[BugFix] Fix parquet array convert bug

### PR DESCRIPTION
**Pull request by meegoo**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #

### CODE REVIEW

File: be/src/exec/vectorized/arrow_to_starrocks_converter.cpp [hunk 1]
```diff
@@ -824,7 +824,7 @@ struct ArrowListConverter {
         auto arrow_base_offset = arrow_offsets_data[0];
         arrow_offsets_data += 1;
         auto start_idx = offsets_layer->size() - 1;
-        offsets_layer->reserve(offsets_layer->size() + layer_num_elements);
+        offsets_layer->resize(offsets_layer->size() + layer_num_elements);
```
Comment (chaoyli) [inline be/src/exec/vectorized/arrow_to_starrocks_converter.cpp:827]: Add a case to the StarRocksTest

**Review by chaoyli — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by meegoo** [comment]
run starrocks_admit_test

**Comment by meegoo** [comment]
run starrocks_admit_test

**Comment by meegoo** [comment]
@mergify backport branch-2.2 branch-2.3

**Pull request merged**